spec.html

```diff
@@ -16635,9 +16634,7 @@
         1. Repeat for each |CaseClause| _C_ in _A_,
           1. If _found_ is *false*, then
             1. Let _clauseSelector_ be the result of CaseSelectorEvaluation of _C_.
-            1. If _clauseSelector_ is an abrupt completion, then
-              1. If _clauseSelector_.[[value]] is ~empty~, return Completion{[[type]]: _clauseSelector_.[[type]], [[value]]: *undefined*, [[target]]: _clauseSelector_.[[target]]}.
-              1. Else, return Completion(_clauseSelector_).
+            1. ReturnIfAbrupt(_clauseSelector_).
```

Comment:
Scratch that, do-expressions probably handle empty completion values themselves. It's also implemented that way in v8 prototype: `L: { 2; !do {break L} }` returns `undefined` in v8.